src/main/java/zmaster587/advancedRocketry/entity/EntityRocket.java

```diff
@@ -1447,32 +1439,29 @@ private void reachSpaceUnmanned()
 				}
 
 				this.setPositionAndUpdate(this.posX, getEntryHeight(destinationDimId), this.posZ);
-				return;
 			}
 
 		}
 		else if(DimensionManager.getInstance().canTravelTo(destinationDimId)) {
 			Vector3F<Float> pos = storage.getDestinationCoordinates(destinationDimId, true);
-			storage.setDestinationCoordinates(new Vector3F<Float>((float)this.posX, (float)this.posY, (float)this.posZ), this.world.provider.getDimension());
+			storage.setDestinationCoordinates(new Vector3F<>((float) this.posX, (float) this.posY, (float) this.posZ), this.world.provider.getDimension());
 			if(pos != null) {
 				this.setInOrbit(true);
 				this.motionY = -this.motionY;
 				this.changeDimension(destinationDimId, pos.x, getEntryHeight(destinationDimId), pos.z);
-				return;
 			}
 			else {
 
 				//Make player confirm deorbit if a player is riding the rocket
 				if(hasHumanPassenger()) {
 					setInFlight(false);
-					pos.y = (float) getEntryHeight(destinationDimId);
+					this.setPositionAndUpdate(this.posX, getEntryHeight(destinationDimId), this.posZ);
```

Comment:
IDEA said this originally caused an NPE due to dereferencing `pos`. I hope my replacement doesn't change the functionality from what was intended